Source/Core/DolphinQt2/Config/FilesystemWidget.cpp

```diff
@@ -4,6 +4,284 @@
 
 #include "DolphinQt2/Config/FilesystemWidget.h"
 
-FilesystemWidget::FilesystemWidget(const GameFile& game) : m_game(game)
+#include <QCoreApplication>
+#include <QFileDialog>
+#include <QHeaderView>
+#include <QMenu>
+#include <QMessageBox>
+#include <QProgressDialog>
+#include <QStandardItemModel>
+#include <QTreeView>
+#include <QVBoxLayout>
+
+#include <future>
+
+#include "DiscIO/DiscExtractor.h"
+#include "DolphinQt2/Resources.h"
+
+static constexpr int ENTRY_PARTITION = Qt::UserRole;
+static constexpr int ENTRY_NAME = Qt::UserRole + 1;
+static constexpr int ENTRY_TYPE = Qt::UserRole + 2;
+
+enum TYPE
```

Comment:
Would it not be possible to do:

```cpp
enum class EntryType
{
  Disc = -2,
  Partition = -1,
  File = 0,
  Dir = 1
};
Q_DECLARE_METATYPE(EntryType);
```

to allow it to be used with QVariant, while still remaining strongly typed?